docs/markdown/Mixing-build-systems.md

```diff
@@ -0,0 +1,51 @@
+# Meson's policy on mixing multiple build systems in one build directory
+
+Meson has been designed with the principle that all dependencies are
+either provided by "the platform" via a mechanism such as Pkg-Config
+or that they are built as Meson subprojects under the main
+project. There are several projects that would like to mix build
+systems, that is, build dependencies in the same build directory in
+the same build directory by having one build system call the other.
+
+This page lists the Meson project's stance on mixing build system. The
+tl/dr version is that while we do provide some functionality for this
+use case, it only works for simple cases. Anything more complex can
+not be made reliable. Thus the support for these use cases is not
+guaranteed to work (in fact it is almost certainly guaranteed to *not*
```

Comment:
I agree with making our policy on mixing build systems clear, but I think this part is a bit too harsh. Would something like `Thus the support for these use cases is unlikely to work.` also work?

Also maybe consider changing `it only works for simple cases` to `it only somewhat reliably works for simple cases`. IMHO if these mechanisms can't at least be classified to work "somewhat reliable" for small/simple projects (which *does* reflect the current status for CMake) then there is no point in having them :)